contracts/crowdsale/price/CapStagedCrowdsale.sol

```diff
@@ -0,0 +1,70 @@
+pragma solidity ^ 0.4.24;
+
+import "../Crowdsale.sol";
+import "../../math/SafeMath.sol";
+import "../../ownership/Ownable.sol";
+
+/**
+ * @title CapStagedCrowdsale
+ * @dev Extension of Crowdsale contract that changes the price of tokens regarding of raised ETH. Each stage of crowdsale is defined in StageLimit (cap limit for the stage) and stageRate (stage rate)
+ * Stages must be inserted from first to last with increasing stageLimit.
+ */
+contract CapStagedCrowdsale is Crowdsale, Ownable {
+  using SafeMath for uint256;
+  using SafeMath for uint;
+
+  struct Stage {
+    uint256 stageLimit;
+    uint256 stageRate;
+  }
+
+
+
+  Stage[] public stages;
+
+  constructor
+  (
+    uint256 _rate,
+    address _wallet,
+    ERC20 _token
+  )
+  Crowdsale(_rate, _wallet, _token) 
+  public
+  {
+  }
+
+  /**
+  * @dev Function for adding stages. Stages must be inserted from first to last
+  * @param stLimit - Stage limit in wei, stRate - limit in weiRaise for the stage
+  */
+  function addStage(uint256 stLimit, uint256 stRate) public onlyOwner {
```

Comment:
we should make it so that stages are added as part of the constructor and then can't be changed; being able to change the stages after the crowdsale starts doesn't make much sense. it's also be useful to instantiate and configure the contract at a single time.

at the very least let's make it so that addStage doesn't work if the crowdsale has started

Reply:
If stages are added during constructor then we must use pragma experimental ABIEncoderV2 which is not in release yet. I could edit it for creator to add first stage, but other stages will still had to be added after contract is published. Because this contract is not necessarily combined with timed crowdsale i can make modifier crowdsaleStarted and we can add stages only before crowdsale is live? 

If you have any suggestion on how to add multiple stages with array in constructor, without using experimental ABIEncoder, i will be more than happy :)

Reply:
i've added two parallel arrays in constructor function so we can now add stages when deploying contract and allow user to add additional ones until crowdsale has started.